backend/src/mail_sender.rs

```diff
@@ -0,0 +1,146 @@
+use std::error::Error;
+use std::fmt::Display;
+use std::fs;
+
+use lettre::message::Mailbox;
+use lettre::message::{Attachment, Body, MultiPart, header::ContentType};
+use lettre::{Address, Message, SmtpTransport, Transport};
+
+#[derive(Debug)]
+pub enum MailSenderError {
+    InvalidEmail,
+    NotInDatabase,
+    InvalidFilePath,
+    NoReceivers,
+    NoFile,
+    CouldntSendEmail,
+}
+
+impl Error for MailSenderError {}
+
+impl Display for MailSenderError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            MailSenderError::InvalidEmail => write!(f, "invalid email"),
+            MailSenderError::NotInDatabase => write!(f, "not in database"),
+            MailSenderError::InvalidFilePath => write!(f, "invalid file path"),
+            MailSenderError::NoReceivers => write!(f, "no receivers"),
+            MailSenderError::NoFile => write!(f, "no file"),
+            MailSenderError::CouldntSendEmail => write!(f, "couldn't send email"),
+        }
+    }
+}
```

Comment:
v bigintu bylo fajn to napsat manuálně, ať si to vyzkoušíš, ale tady si spíš importuj `thiserror` crate a vygeneruj to tim atributem